br/pkg/aws/ebs.go

```diff
@@ -326,6 +326,36 @@ func (e *EC2Session) EnableDataFSR(meta *config.EBSBasedBRMeta, targetAZ string)
 
 // waitDataFSREnabled waits FSR for data volume snapshots are all enabled
 func (e *EC2Session) waitDataFSREnabled(snapShotIDs []*string, targetAZ string) error {
+	// Record current time
+	start := time.Now()
+
+	//  get the maximum size of volumes, in GiB
+	var maxVolumeSize int64 = 0
+	resp, err := e.ec2.DescribeSnapshots(&ec2.DescribeSnapshotsInput{SnapshotIds: snapShotIDs})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(resp.Snapshots) <= 0 {
+		return errors.Errorf("specified snapshot [%s] is not found", *snapShotIDs[0])
+	}
+
+	for _, s := range resp.Snapshots {
+		if *s.VolumeSize > maxVolumeSize {
+			maxVolumeSize = *s.VolumeSize
+		}
+	}
+
+	// Calculate the time in minutes to fill 1.0 credit according to
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-fast-snapshot-restore.html#volume-creation-credits
+	fillElapsedTime := 60.0 / (min(10, 1024.0/maxVolumeSize))
+
+	// We have to sleep for at least fillElapsedTime minutes in order to make credits are filled to 1.0
+	// Let's heartbeat every 5 minutes
+	for time.Since(start) <= time.Duration(fillElapsedTime)*time.Minute {
+		log.Info("FSR enablement is ongoing, going to sleep for 5 minutes...")
+		time.Sleep(5 * time.Minute)
+	}
+
 	// Create a map to store the strings as keys
 	pendingSnapshots := make(map[string]struct{})
 
```

Comment:
Nit question for https://github.com/pingcap/tidb/blob/7430f05c3f4a29a2f40aed38dac035956d5852e8/br/pkg/aws/ebs.go#L381. 

Why not check for "enabled" and remove those snapshots from the pending list. There should be much fewer enabled snapshots. Also seems simpler.

Reply:
I am afraid there could some conflict fsr operation to those snapshots in case.

Reply:
What does this mean. What conflicts?